library/alloc/src/collections/vec_deque/iter.rs

```diff
@@ -142,28 +112,36 @@ impl<'a, T> Iterator for Iter<'a, T> {
 impl<'a, T> DoubleEndedIterator for Iter<'a, T> {
     #[inline]
     fn next_back(&mut self) -> Option<&'a T> {
-        if self.tail == self.head {
-            return None;
+        match self.i2.next_back() {
+            Some(val) => Some(val),
+            None => {
+                // most of the time, the iterator will either always
+                // call next(), or always call next_back(). By swapping
+                // the iterators once the first one is empty, we ensure
```

Comment:
nit: this is copy-pasted, but I think it'd be slightly better to say "once the second one is empty" for these `_back` versions.

Reply:
Depends on if you interpret "the first iterator" as always referring to `i1`, or as referring to "the iterator that we polled first". I can change it though, to avoid confusion.